tests/e2e/default/reports/search-reports.wdio-spec.js

```diff
@@ -1,114 +1,76 @@
+    pregnancyFactory.build({ fields: { patient_id: sittuPerson.patient_id, case_id: 'case-12' } }),
+    pregnancyFactory.build({ fields: { patient_id: potuPerson.patient_id, case_id: 'case-12' } }),
+  ];
+  let reportDocs;
 
-const REPORT2 = {
-  _id: 'REF_REF_V2',
-  form: 'RR',
-  type: 'data_record',
-  from: PHONE_NO,
-  fields: {
-    patient_id: potuPerson.patient_id
-  },
-  sms_message: {
-    message_id: 23,
-    from: PHONE_NO,
-    message: `1!RR!${potuPerson.patient_id}`,
-    form: 'RR',
-    locale: 'en'
-  },
-  reported_date: moment().subtract(10, 'minutes').valueOf()
-};
-
-const docs = [...places.values(), sittuHospital, sittuPerson, potuHospital, potuPerson, REPORT1, REPORT2];
-
-describe('Test Reports Search Functionality', async () => {
   before(async () => {
-    await utils.saveDocs(docs);
+    await utils.saveDocs([ sittuHospital, sittuPerson, potuHospital, potuPerson ]);
+    reportDocs = await utils.saveDocs(reports);
     await loginPage.cookieLogin();
-    await commonPage.goToReports();
   });
 
-  it('search by NON empty string should display results with contains match and then clears', async () => {
-    // Waiting for initial load
-    await reportsPage.getAllReportsText();
+  it('should return results matching the search term and then return all data when clearing search', async () => {
+    const [ sittuSMSPregnancy, potuSMSPregnancy, sittuPregnancy, potuPregnancy ] = reportDocs;
+    await commonPage.goToReports();
+    // Asserting first load reports
+    expect((await reportsPage.reportsListDetails()).length).to.equal(reportDocs.length);
 
-    // Searching by keyword
     await searchPage.performSearch('sittu');
-    let allLHSContacts = await reportsPage.getAllReportsText();
-    expect(allLHSContacts.sort()).to.deep.equal([sittuPerson.name]);
+    await commonPage.waitForLoaders();
+    expect((await reportsPage.reportsListDetails()).length).to.equal(2);
+    expect(await (await reportsPage.reportByUUID(sittuSMSPregnancy.id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(sittuPregnancy.id)).isDisplayed()).to.be.true;
 
-    // Clearing
     await searchPage.clearSearch();
-    allLHSContacts = await reportsPage.getAllReportsText();
-    expect(allLHSContacts.sort()).to.deep.equal([potuPerson.name, sittuPerson.name]);
+    expect((await reportsPage.reportsListDetails()).length).to.equal(reportDocs.length);
+    expect(await (await reportsPage.reportByUUID(sittuSMSPregnancy.id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(potuSMSPregnancy.id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(sittuPregnancy.id)).isDisplayed()).to.be.true;
+    expect(await (await reportsPage.reportByUUID(potuPregnancy.id)).isDisplayed()).to.be.true;
+  });
+
+  it('should return results when searching by case_id', async () => {
+    const sittuPregnancy = reportDocs[2];
+    const potuPregnancy = reportDocs[3];
```

Comment:
```suggestion
const [, sittuPregnancy, potuPregnancy] = reportDocs;
```

Reply:
I thought about this, but I don't like that much to have "empties". In this case, we need two: 
`const [ , , sittuPregnancy, potuPregnancy ]`
Currently, it's equally as readable, just more lines. But if you feel strongly about this, I can change it :)